src/mbedtls_context.cpp

```diff
@@ -587,7 +588,30 @@ namespace sockpp {
     {
         if (!root_cert_locator_)
             return -1;
-        string certData((const char*)child->raw.p, child->raw.len);
+        
+        // Construct the cert chain including all intermediates in PEM format:
+        string certData;
+        for (auto crt = child; crt; crt = crt->next) {
+            size_t olen = 0;
+            unsigned char buf[4096];
```

Comment:
Is it OK to use a fixed size buffer? What if the cert is bigger?

Reply:
I have just changed the buffer size to 10K to match what LiteCore is using in mbedUtils.cc.

Reply:
I have review the code in LiteCore that uses mbedUtils.convertToPEM(), and I have found that there are not many code paths that uses that function, especially the paths that convert cert into PEM format. To be sure that we do have the problem, I have changed the code to use a dynamic buffer and retry again if the initial buffer size is too small.